v2 release notes (#1526)
stoplightio/stoplight.io: netlify/blog-posts/platform-and-studio-v2.md

---
path: /blog/stoplight-v2-0-0-release
tags:
  - blog
publishedDate: 2020-09-02T12:26:13.173Z
author: Phil Sturgeon
title: Stoplight v2.0 Release

> subtitle: Studio and Platform

Reviewer: Can we make the subtitle to have a sneak peek of the features instead of this?

listSubtitle: Studio and Platform v2.0 Releases | Stoplight API Intersection
image: /images/changelog-stock.jpg
includeToc: true

> actionBar:

Reviewer 2: Can you try removing ` ctas: []` from `actionBar`? For some reason it doesn't show the full CTA, but it should be defaulting to the one in all other blogs (example: https://deploy-preview-1526--stoplightio.netlify.app/blog/is-api-planning-the-same-thing-as-api-design/). If that doesn't fix it, let me know.

  ctas: []
  enabled: false
meta:
  description: 'v2.0 Releases for Stoplight Studio and Platform'
  url: https://stoplight.io/blog/stoplight-v2-0-0-release/
  favicon: /images/mark_light_bg.png
  robots: 'index, follow'
  title: Studio and Platform v2.0 Releases | Stoplight API Intersection
  image: /images/changelog-stock.jpg
  twitter:
    description: 'v2.0 Releases for Stoplight Studio and Platform'
    title: Studio and Platform v2.0 Releases | Stoplight API Intersection
    image: /images/changelog-stock.jpg
    username: '@stoplightio'
---

Last month we announced [Stoplight Platform](https://stoplight.io/blog/stoplight-platform/) to the world, a revolutionary ecosystem that lets you handle API Design, Documentation, and API Mocking. Unlike other tooling, all this functionality, and the API descriptions and documentation can live in your existing version control system, along with the source code. The new approach handles all this whilst keeping things easy enough for your whole team to use, not just the techie folks.

Now that we've had some time to really polish things, we're ready to release major versions for two parts of our ecosystem:

> - [Stoplight Studio Desktop v2.0](#studio-desktop-v2-0)

Reviewer 2: These links do not work

- [Stoplight Platform v2.0](#platform-v2-0)

> For those of you new to [Studio Desktop](/studio), it's is a standalone application for Windows, macOS, and Linux, which allows you to edit and manage OpenAPI, JSON Schema, AsyncAPI, and Markdown documents locally. Studio Desktop works with files on your computer, has built in Git functionality, runs local mocking servers for you, and it's totally free. You can use it with any Git repository or local directory you want, whether it has anything to do with Stoplight or not. 

Reviewer: ```suggestion
For those of you new to [Studio Desktop](/studio), it's is a standalone application for Windows, macOS, and Linux, which allows you to edit and manage OpenAPI, JSON Schema, AsyncAPI, and Markdown documents locally. Studio Desktop works with files on your computer, has built-in Git functionality, runs local mocking servers for you, and it's totally free. You can use it with any Git repository or local directory you want, whether it has anything to do with Stoplight or not. 
```

> For those of you new to [Studio Desktop](/studio), it's is a standalone application for Windows, macOS, and Linux, which allows you to edit and manage OpenAPI, JSON Schema, AsyncAPI, and Markdown documents locally. Studio Desktop works with files on your computer, has built in Git functionality, runs local mocking servers for you, and it's totally free. You can use it with any Git repository or local directory you want, whether it has anything to do with Stoplight or not. 

Reviewer 2: ```suggestion
For those of you new to [Studio Desktop](/studio), it's a standalone application for Windows, macOS, and Linux, which allows you to edit and manage OpenAPI, JSON Schema, AsyncAPI, and Markdown documents locally. Studio Desktop works with files on your computer, has built in Git functionality, runs local mocking servers for you, and it's totally free. You can use it with any Git repository or local directory you want, whether it has anything to do with Stoplight or not. 
```

> For those of you new to [Studio Desktop](/studio), it's is a standalone application for Windows, macOS, and Linux, which allows you to edit and manage OpenAPI, JSON Schema, AsyncAPI, and Markdown documents locally. Studio Desktop works with files on your computer, has built in Git functionality, runs local mocking servers for you, and it's totally free. You can use it with any Git repository or local directory you want, whether it has anything to do with Stoplight or not. 

Reviewer 2: What do you think about including a link up here as well to where someone can download? Just in case they do not read the entire post?

Author: Good idea, done!

For those of you new to Platform, it's what we call the rest of the API design and discovery ecosystem, which is [free to use](https://stoplight.io/welcome) in our cloud or can be self-hosted on your own servers with an [Enterprise license](https://stoplight.io/enterprise/).

Let's talk about Platform v2.0 first, because it's got some of the biggest changes!

## Platform v2.0

To use the latest version of the Stoplight Platform, you can [create a free Workspace](https://stoplight.io/welcome) in our cloud environment or install the "2.0.0" image with your [Enterprise license](https://stoplight.io/enterprise/).

### Workspaces

Workspaces are a handy way to group API projects across your organization, to promote visibility, and reuse of artifacts. We at Stoplight recommend making a single workspace for your organization and using roles and permissions to effectively manage your workflows.

![](/images/v2-meta-workspace.png)

Read more about [workspaces](https://meta.stoplight.io/docs/platform/2.-workspaces/a.creating-a-workspace.md).

### Automatic Publishing

> In the past, to update your docs, somebody would have needed to pull changes in Studio Desktop or Studio Web, and manually click a publish button. Now that manual step is now longer needed. Webhooks from GitHub, GitLab, and BitBucket can all be used to automatically update your documentation, Explorer, and mock servers, and even works with multiple branches.

Reviewer: ```suggestion
In the past, to update your docs, somebody would have needed to pull changes in Studio Desktop or Studio Web, and manually click a publish button. Now that manual step is no longer needed. Webhooks from GitHub, GitLab, and BitBucket can all be used to automatically update your documentation, Explorer, and mock servers, and even works with multiple branches.
```

> In the past, to update your docs, somebody would have needed to pull changes in Studio Desktop or Studio Web, and manually click a publish button. Now that manual step is now longer needed. Webhooks from GitHub, GitLab, and BitBucket can all be used to automatically update your documentation, Explorer, and mock servers, and even works with multiple branches.

Reviewer 2: ```suggestion
In the past, to update your docs, somebody would have needed to pull changes in Studio Desktop or Studio Web, and manually click a publish button. Now that manual step is no longer needed. Webhooks from GitHub, GitLab, and Bitbucket can all be used to automatically update your documentation, Explorer, and mock servers, and even works with multiple branches.
```

![](/images/blog/studio-and-platform-v2-0/automation.png)

> Stoplight Platform takes the burden of publishing off of your team, and gets robots on the task [using Webhooks, or Continuous Integration](https://meta.stoplight.io/docs/platform/2.-workspaces/g.automating-publishing.md). Why do things when you could not do things.

Reviewer 2: ```suggestion
Stoplight Platform takes the burden of publishing off of your team, and gets robots on the task [using Webhooks, or Continuous Integration](https://meta.stoplight.io/docs/platform/2.-workspaces/g.automating-publishing.md). Why do things when you could not do things?
```

> Stoplight Platform takes the burden of publishing off of your team, and gets robots on the task [using Webhooks, or Continuous Integration](https://meta.stoplight.io/docs/platform/2.-workspaces/g.automating-publishing.md). Why do things when you could not do things.

Reviewer 2: ```suggestion
Stoplight Platform takes the burden of publishing off of your team, and gets robots on the task [using Webhooks, or Continuous Integration](https://meta.stoplight.io/docs/platform/2.-workspaces/g.automating-publishing.md). Why do things when you could not do things?
```

### Multiple Branch Support

One of the main goals for Stoplight is not to impose workflows on your organization, and for those using Git we want to support whatever branching model is in use.

> Maybe a repository follows [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) with `master`, `develop` and `feature/` branches. Maybe another project has versions spread across multiple version branches, like `v3` and `v4`. Publishing can be enabled for multiple branches, so that master and develop could both have documentation.

Reviewer 2: ```suggestion
Maybe a repository follows [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) with `master`, `develop` and `feature/` branches. Maybe another project has versions spread across multiple version branches, like `v3` and `v4`. Publishing can be enabled for multiple branches, so that `master` and `develop` could both have documentation.
```

> Maybe a repository follows [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) with `master`, `develop` and `feature/` branches. Maybe another project has versions spread across multiple version branches, like `v3` and `v4`. Publishing can be enabled for multiple branches, so that master and develop could both have documentation.

Reviewer 2: ```suggestion
Maybe a repository follows [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) with `master`, `develop` and `feature/` branches. Maybe another project has versions spread across multiple version branches, like `v3` and `v4`. Publishing can be enabled for multiple branches, so that `master` and `develop` could both have documentation.
```

![](/images/blog/studio-and-platform-v2-0/enable-publishing.png)

Branches may also be given Display Names to make them more clear to those viewing the documentation, maybe `master` could be nicknamed "Stable" and `develop` "Development".

Read more about [branch management](https://meta.stoplight.io/docs/platform/2.-workspaces/h.branch-management.md).

### Hierarchical Explorer

> A powerful full-text search tool for exploring all of the API design assets in your Workspace. The new hierarchical view allows you to quickly traverse the contents of your API projects.

Reviewer 2: I don't think this is a complete sentence

![](/images/blog/studio-and-platform-v2-0/explorer.png)

Check out some background on [Explorer](https://meta.stoplight.io/docs/platform/5.-governance/bb.exploring-your-api-projects.md).

### Theming and Branding

Pick one of the preset themes or provide your own colors to fully customize your workspace's branding!

> ![](/images/blog/studio-and-platform-v2-0/theming.gif)

Reviewer: Been adding an example from one of our customer in intercom. Could be helpful to see it in action: https://staircase.stoplight.io

### Expanded Git Integrations

Stoplight supports Team Foundation Server 2018 (v4.1), which has since been renamed to [Azure DevOps Server](https://docs.microsoft.com/en-us/azure/devops/server/tfs-is-now-azure-devops-server?view=azure-devops).

## Studio Desktop v2.0

To download the latest version of Studio Desktop, visit the download page [here](https://github.com/stoplightio/studio/releases). Studio Web is included in the latest release of Stoplight Platform.

### Automatic Publishing

With Publishing now handled in Stoplight Platform, there's no need for the Publish button anymore, so we've gone ahead and chucked that out to clear up the UI for you.

The new docs are published to a new and improved URL for Stoplight.io users, so take a look at how [publishing works in v1 vs v2](https://meta.stoplight.io/docs/studio/docs/Documentation/06-publishing.md).

### Supports on-premise VCS providers

> In the v1, all Git requests were proxied through the Stoplight API, which required you to login to your Stoplight account and was limited to Github. Now you can use personal access tokens to authenticate with any VCS provider.

Reviewer 2: ```suggestion
In the v1, all Git requests were proxied through the Stoplight API, which required you to login to your Stoplight account and was limited to GitHub. Now you can use personal access tokens to authenticate with any VCS provider.
```

### Performance improvements

Performance has seen a whole bunch of improvements in v2.0, with two areas seeing the most improvements:

1. Project load time 
2. Git operations, such as Clone, and Commit & Push

Various other efforts have been made to speed up linting, and we now internally work with "bundled" OpenAPI instead of "dereferenced" OpenAPI, which means we can handle much larger files before things start to slow down.

### Linter Documentation

Warnings and Errors in the Problems panel (linter messages from Spectral) used to exist in a bit of a vacuum. They'd show up, with one line saying what the problem is, and hopefully people would know what that meant. 

> To make things more easy to understand [we've added](https://github.com/stoplightio/studio/issues/445) links to the ruleset documentation next to each linter message. The [ruleset documentation has always existed](https://meta.stoplight.io/docs/spectral/docs/reference/openapi-rules.md), but if you didn't know it was there it wasn't much use! Now we link to each relevant part of the documentation for every linter rule. 

Reviewer: ```suggestion
To make things easier to understand [we've added](https://github.com/stoplightio/studio/issues/445) links to the ruleset documentation next to each linter message. The [ruleset documentation has always existed](https://meta.stoplight.io/docs/spectral/docs/reference/openapi-rules.md), but if you didn't know it was there it wasn't much use! Now we link to each relevant part of the documentation for every linter rule. 
```

> To make things more easy to understand [we've added](https://github.com/stoplightio/studio/issues/445) links to the ruleset documentation next to each linter message. The [ruleset documentation has always existed](https://meta.stoplight.io/docs/spectral/docs/reference/openapi-rules.md), but if you didn't know it was there it wasn't much use! Now we link to each relevant part of the documentation for every linter rule. 

Reviewer 2: ```suggestion
To make things easier to understand [we've added](https://github.com/stoplightio/studio/issues/445) links to the ruleset documentation next to each linter message. The [ruleset documentation has always existed](https://meta.stoplight.io/docs/spectral/docs/reference/openapi-rules.md), but if you didn't know it was there it wasn't much use! Now we link to each relevant part of the documentation for every linter rule. 
```

TODO show branch nicknames in documents

Read more about [banch management](https://meta.stoplight.io/docs/platform/2.-workspaces/h.branch-management.md).

### Custom Domains

TODO do we have any docs for how this works for on prem?

## Studio Desktop v2.0

To download the latest version of Studio Desktop, visit the download page [here](https://github.com/stoplightio/studio/releases). Studio Web is included in the latest release of Stoplight Platform.

### Auto-Publishing

In the past Studio users would click the Publish button in order to update documentation, mock servers, etc. This required a Stoplight login to do, and could only Publish to our [hosted Documentation](https://stoplight.io/docs/), so it was a bit confusing for users who were using Studio Desktop but not using Stoplight. 

With the new Automated Publishing offered in Stoplight Platform, there is no longer any need for the Publish button in Studio. 

See how [publishing with Studio](https://meta.stoplight.io/docs/studio/docs/Documentation/06-publishing.md) works in v2.x.

### Offline First

> focused on working with local files on your computer

Author: @lottamus can you help me expand on this a bit? what makes this a v2 thing?

TODO Chris this was listed as new but... wasn't it always?

### Login has been removed

> Either this is a new thing we're announcing, which is not really a feature. whats the sales pitch?

Author: I either need to find a way to make this sound positive, or cushion the blow, or make it at least be not confusing.

There's a chance we're not removing it, I spoke to marc about how this would not be ideal with many 2.x features needing a login.

You can now use personal access tokens to authenticate with your VCS provider which supports on-premise git providers.

### Performance improvements

> - project activation time is reduced, etc.

Author: Any numbers of rough examples of this so I can wow people instead of just hand waving?

- git operations such as clone & commit/push are faster

- Link to Spectral ruleset docs (core ruleset or other rulesets) https://github.com/stoplightio/studio/issues/445

### UI Changes

> Spectral and Mocking bottom panel now appears in the right panel

Author: Meh? Do we care? What about this is subtantially better.
